src/openforms/registrations/contrib/zgw_apis/plugin.py

```diff
@@ -117,12 +165,33 @@ def validate(self, attrs: dict[str, Any]) -> dict[str, Any]:
                     code="invalid",
                 )
 
-        if not ("medewerker_roltype" in attrs and "zaaktype" in attrs):
-            return attrs
-
         # We know it exists thanks to the previous check
         group_config = ZGWRegistration.get_zgw_config(attrs)
 
+        # Make sure the property (eigenshap) related to the zaaktype exists
+        if connections := attrs.get("variables_properties"):
+            with get_catalogi_client(group_config) as client:
+                eigenshappen = client.list_eigenshappen(
+                    zaaktype=attrs.get("zaaktype") or group_config.zaaktype
+                )
+                retrieved_eigenshappen = {
+                    eigenshap["naam"]: eigenshap["url"] for eigenshap in eigenshappen
+                }
+
+                for conn in connections:
+                    if conn["eigenshap"] not in retrieved_eigenshappen:
+                        raise serializers.ValidationError(
+                            {
+                                "variables_properties": _(
+                                    "Could not find a property with the name '{property_name}' related to the zaaktype."
+                                ).format(property_name=conn["eigenshap"])
+                            },
+                            code="invalid",
+                        )
```

Comment:
Can we collect all the errors and then only raise an error once? If a user makes two typo's, ideally you get that reported back once instead of fixing the first, and then getting another validation error for the second.